### DISCUSSION TITLE
HelmRelease version: 1.3 quietly fails - chart version could not be found

### DISCUSSION THREAD

**Discussion by cer** [fluxcd/flux2#3791] [General]
* The Helm CLI was happy to install chart version 1.3
* When I created a HelmRelease to do the same, the HelmChart complained that version 1.3 could not be found.
* I'm guessing that this is because the version is not semver compliant: `x.y.z`
* e.g. republishing the chart as `1.3.0` was fine

It would be helpful if Flux complained that the version was not semver compliant.

**Comment by stefanprodan** [comment]
This is definitely a bug in Helm as it should validate the version at push time. I agree that we need to improve the error message on our side as well, but this can be costly as we would have to parse all entries in the Helm repo index for bogus semver. Thanks for reporting this Chris.

**Reply by cer** [reply]
Interesting. Helm didn't complain.

Can't Flux simply verify that a HelmRelease's chart version is Semver compliant?

```yaml
kind: HelmRelease
..
spec:
  chart:
    spec:
      chart: ...
      version: ...
```

**Reply by stefanprodan** [reply]
We verify that it’s a valid semver range and it is, it means latest patch of 1.3

**Reply by hiddeco** [reply]
Likely indirectly related to https://github.com/helm/helm/issues/11107

**Reply by stefanprodan** [reply]
IMO we shouldn't do anything for this in Flux. It's Helm responsibility to validate the chart version at build time.
